src/react/reconcilation.js

```diff
@@ -904,6 +904,45 @@ export class Reconciler {
     return value;
   }
 
+  _resolveAbstractLogicalValue(
+    componentType: Value,
+    value: AbstractValue,
+    context: ObjectValue | AbstractObjectValue,
+    evaluatedNode: ReactEvaluatedNode
+  ) {
+    let [leftValue, rightValue] = value.args;
+    let operator = value.kind;
+
+    invariant(leftValue instanceof AbstractValue);
+    if (operator === "||") {
+      return this._resolveAbstractConditionalValue(
+        componentType,
+        leftValue,
+        leftValue,
+        rightValue,
+        context,
+        evaluatedNode
+      );
+    } else {
+      // When we have &&, for now we don't support inlining the left side of the logical operation
+      // and instead only attempt to inline the right side. This is the most common case in React
+      // where && is used as shorthand for a conditional check given left side happens, use right.
+      // Furthermore, as the right side might have generator entries, we need to evaluate its
+      // effects and wrap it in a conditional (the condition is always the left side).
+      let effects = Path.withCondition(leftValue, () => {
```

Comment:
If leftValue evaluates with side-effects, these effects will be invisible to the _resolveDeeply call below. Perhaps such code should not be allowed, but in that case a check will be needed. So either way, leftValue should be evaluated before rightValue.

Reply:
How would I evaluate `leftValue`? I thought by this point it was already evaluated?

Reply:
I see. What exactly does __resolveDeepty do and why do you use evaluateForEffects on it?

Reply:
It takes the value and tries to unfold it if it’s a valid React value (ReactElement, array, abstract value) that allows us to generate a new value from it. Doing this might create generator entries that need to be wrapped in a condition (otherwise we get runtime errors)